tiledb/tests/test_attribute.py

```diff
@@ -252,3 +252,52 @@ def test_modify_attribute_in_schema(self):
                 assert "can't set attribute" in str(exc.value)
             else:
                 assert "object has no setter" in str(exc.value)
+
+    def test_wkt_attribute(self):
+        A = np.array(
+            ["POINT (30 10)", "POLYGON ((3 1, 4 5, 2 2, 1 2, 3 1))"],
+            dtype="S",
+        )
+
+        dom = tiledb.Domain(tiledb.Dim(domain=(0, 1), tile=2))
+        att = tiledb.Attr(dtype="wkt", var=True)
+
+        schema = tiledb.ArraySchema(dom, (att,))
+
+        tiledb.DenseArray.create(self.path("foo"), schema)
+        with tiledb.DenseArray(self.path("foo"), mode="w") as T:
+            T[:] = A
+
+        # read back the data
+        with tiledb.DenseArray(self.path("foo"), mode="r") as T:
+            for i in range(2):
+                assert_array_equal(T[:][i].tobytes(), A[i])
```

Comment:
I think from the Python API, I would expect "WKT" to behave more like a string type than a blob type. For example, I would expect back the same thing we get from an ascii datatype:
```
array([b'POINT (30 10)', b'POLYGON ((3 1, 4 5, 2 2, 1 2, 3 1))'],
      dtype=object)
```
instead of 
```
array([array([b'P', b'O', b'I', b'N', b'T', b' ', b'(', b'3', b'0', b' ', b'1',
              b'0', b')'], dtype='|S1')                                        ,
       array([b'P', b'O', b'L', b'Y', b'G', b'O', b'N', b' ', b'(', b'(', b'3',
              b' ', b'1', b',', b' ', b'4', b' ', b'5', b',', b' ', b'2', b' ',
              b'2', b',', b' ', b'1', b' ', b'2', b',', b' ', b'3', b' ', b'1',
              b')', b')'], dtype='|S1')                                        ],
      dtype=object)
```

That being said, it might be worth asking the geospatial team to confirm this is the behavior they would expect as well.

Reply:
Solved.

---
- **np.dtype("S")** represents a string of fixed length but without a specified length. Length of the string is determined by the data provided.
- **np.dtype("S1")** This represents a string of fixed length where the length is explicitly set to 1 byte.